### PR TITLE
Title field XSS vulnerability solution.

### PR DESCRIPTION
**Pull request by Nould**
This may be solution of a ticket's Title field XSS vulnerabilty in Backoffice and Portal. Backoffice's part is awful but it works.

### CODE REVIEW

File: application/cmdbabstract.class.inc.php [hunk 1]
```diff
@@ -2201,7 +2201,8 @@ public static function GetFormElementForField($oPage, $sClass, $sAttCode, $oAttD
 						{
 							$oPage->add_ready_script(
 								<<<EOF
-								$('#{$iId}').qtip( { content: $('#{$iId}').val(), show: 'mouseover', hide: 'mouseout', style: { name: 'dark', tip: 'bottomLeft' }, position: { corner: { target: 'topLeft', tooltip: 'bottomLeft' }, adjust: { y: -15}} } );
+								var escapedVal = $('<div/>').text($('#{$iId}').val()).html();
```
Comment (steffunky) [inline application/cmdbabstract.class.inc.php]:
```suggestion
								var sEscapedVal = $('<div/>').text($('#{$iId}').val()).html();
```
Our naming convention defines the variable type with a letter prefix, if you could add this little change!

File: application/cmdbabstract.class.inc.php [hunk 2]
```diff
@@ -2201,7 +2201,8 @@ public static function GetFormElementForField($oPage, $sClass, $sAttCode, $oAttD
...
+								$('#{$iId}').qtip( { content: escapedVal, show: 'mouseover', hide: 'mouseout', style: { name: 'dark', tip: 'bottomLeft' }, position: { corner: { target: 'topLeft', tooltip: 'bottomLeft' }, adjust: { y: -15}} } );
```
Comment (steffunky) [inline application/cmdbabstract.class.inc.php]:
```suggestion
								$('#{$iId}').qtip( { content: sEscapedVal, show: 'mouseover', hide: 'mouseout', style: { name: 'dark', tip: 'bottomLeft' }, position: { corner: { target: 'topLeft', tooltip: 'bottomLeft' }, adjust: { y: -15}} } );
```

File: datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig [hunk 3]
```diff
@@ -118,7 +118,7 @@
 
                             // Preparing the cell data
                             cellElem = (itemActions.length > 0) ? $('<a></a>') : $('<span></span>');
-                            cellElem.html(row.attributes[att_code].value);
+                            cellElem.text(row.attributes[att_code].value);
```
Comment (steffunky) [inline datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig]: We didn't quite understand what would be the vulnerability here, when we breakpoint on this variable, content seems to be already escaped, could you provide us some information on how you exploit this vulnerability ?

Comment (Nould) [reply datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig]: I'm sorry, I probably found this problem with XSS vulnerability in some of previous versions. Now i seems to be ok. I changed it back.

**Comment by Molkobain** [comment]
Nevermind, I didn't see that @Nould answered. All good to me :)

File: datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig (hunk 3 above)
Comment (Nould) [reply datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig]: There was a problem in version 2.5.0-3935, where I tried it. But when I tested it with sources from develop branch, it seems to be ok.

Comment (Hipska) [reply datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig]: So, if it is also okay on 2.6 branch, then this PR can be closed as it doesn't fix anything?

Comment (Nould) [reply datamodels/2.x/itop-portal-base/portal/src/views/bricks/manage/layout-table.html.twig]: Only the layout-table.html.twig changes were useless. The cmdbabstract.class.inc.php changes fixes XSS vulnerability in Backoffice, so they should be merged.

**Comment by Molkobain** [comment]
Hi @Nould ,

Could you please stash all the commits into one? As they are all part of the same fix, there is no need for several commits, it will help keep the change log clean. Thanks!

**Comment by Nould** [comment]
Done.

**Comment by steffunky** [comment]
Rebased and merged! Thanks for your contribution @Nould !

**Pull request merged**